bc2wrap/src/lib.rs

```diff
@@ -16,13 +16,16 @@ pub fn serialize<T: serde::Serialize + ?Sized>(
 }
 
 /// wrapper around bincode::serde::decode_from_slice that discards the length info and uses the legacy config
-/// (using bincode v2 underneath)
-pub fn deserialize<T: serde::de::DeserializeOwned>(
+/// (using bincode v2 underneath).
+/// This is unsafe as it does not limit the size of the deserialized object and thus may lead to OOM errors.
+pub fn deserialize_unsafe<T: serde::de::DeserializeOwned>(
```

Comment:
I am wondering if there is actually anywhere, besides test code, where we want to do this? If not, then maybe we want to hide this behind the `test` or `testing` flag